pkg/api/annotations.go

```diff
@@ -52,9 +53,14 @@ func (hs *HTTPServer) GetAnnotations(c *models.ReqContext) response.Response {
 		dq := models.GetDashboardQuery{Uid: query.DashboardUid, OrgId: c.OrgID}
 		err := hs.DashboardService.GetDashboard(c.Req.Context(), &dq)
 		if err != nil {
-			return response.Error(http.StatusBadRequest, "Invalid dashboard UID in the request", err)
+			if hs.Features.IsEnabled(featuremgmt.FlagDashboardsFromStorage) {
```

Comment:
should we differentiate between `dashboards.ErrDashboardNotFound` and other errors for this case?